File: cpp/se/PeriodUnit.cpp
#include "se/PeriodUnit.hpp"

std::vector<const EnumType *> PeriodUnit::values;

const PeriodUnit* PeriodUnit::MONTHS()
{
	static const PeriodUnit unit("m");
	return &unit;
}

const PeriodUnit* PeriodUnit::YEARS()
{
	static const PeriodUnit unit("y");
	return &unit;
}

PeriodUnit::PeriodUnit(const std::string &description)
    : EnumType(values, description)
{ }

const PeriodUnit* PeriodUnit::value (const std::string &name)
{
    try
    {
        return (const PeriodUnit *)EnumType::value (name, values);
    }
    catch (IllegalArgException)
    {
        return YEARS();
    }
}

void PeriodUnit::init()
{
	MONTHS();
	YEARS();
}

File: src/main/java/com/ausregistry/jtoolkit2/se/PeriodUnit.java
package com.ausregistry.jtoolkit2.se;

/**
 * Enumeration of units supported by EPP for period elements.
 */
public enum PeriodUnit {
    MONTHS("m"), YEARS("y");

    private String desc;

    PeriodUnit(String description) {
        desc = description;
    }

    public String toString() {
        return desc;
    }

    public static PeriodUnit value(String name) {
        if (name == null) {
            return PeriodUnit.YEARS;
        }

        for (PeriodUnit unit : PeriodUnit.values()) {
            if (unit.toString().equals(name)) {
                return unit;
            }
        }

        return PeriodUnit.YEARS;
    }
}

